qiskit/visualization/pulse/matplotlib.py

```diff
@@ -67,10 +67,17 @@ def add_instruction(self, start_time, pulse):
             pulse (Instruction): Instruction object to be added
         """
 
+        if pulse.command is not None:
+            # TODO: This will be deprecated
+            operator = pulse.command
+        else:
+            # TODO: Add if statement for Pulse: Qiskit-terra #3936 (to store pulse.pulse)
```

Comment:
What needs to go here?

Reply:
something like
```python

        if pulse.command is not None:
            # TODO: This will be deprecated
            operator = pulse.command
        else:
            if isinstance(pulse, Play):
                operator = pulse.pulse
            else:
                operator = pulse

        if start_time in self.pulses.keys():
            self.pulses[start_time].append(operator)
        else:
            self.pulses[start_time] = [operator]
```
it access to `.sample` property but `Play` instruction doesn't offer this.
I think argument name is better to be changed to something like `instruction`.

there are two approaches

1. merge this first and @lcapelluto add code
2. merge #3936 first and I update the code

Reply:
Ahh I should just add this to my PR, since that's what is going to break it, right? Thank you for seeing this!